hexrd/ui/calibration/wppf_options_dialog.py

```diff
@@ -420,6 +421,136 @@ def create_wppf_params_object(self):
             params.add(**kwargs)
         return params
 
+    def add_material_parameters(self):
+        """
+        @AUTHOR:    Saransh Singh, Lawrence Livermore National Lab
+        @DATE:      02/03/2021 SS 1.0 original
+        @DETAILS:   a simple function to add the material parameters
+        from the list of material file. this depends on which
+        method i chosen. for the LeBail class the parameters
+        added are the minimum set of lattice parameters. For 
+        the Rietveld class, the lattice parameters, fractional
+        coordinates, occupancy and debye waller factors are 
+        added.
+        """
+        method = self.wppf_method
+
+        for x in self.selected_materials:
+            mat = HexrdConfig().material(x)
+
+            # if method is LeBail
+            if method == 'LeBail':
+                """
+                add lattice parameters
+                """
+                lp = np.array(mat.planeData.lparms)
+                rid = list(_rqpDict[mat.unitcell.latticeType][0])
+
+                name = _lpname[rid]
+
+                for i, (n, l) in enumerate(zip(name, lp)):
+                    nn = p+'_'+n
+
+                    """
+                    first 3 are lengths, next three are angles
+                    """
+                    if(rid[i] <= 2):
+                        self.params.add(nn, value=l, lb=l-0.05,
+                                   ub=l+0.05, vary=False)
+                    else:
+                        self.params.add(nn, value=l, lb=l-1.,
+                                   ub=l+1., vary=False)
+
+            elif method == 'Rietveld':
+                """
+                add lattice parameters
+                """
+                lp = np.array(mat.planeData.lparms)
+                rid = list(_rqpDict[mat.unitcell.latticeType][0])
+
+                name = _lpname[rid]
+                for i, (n, l) in enumerate(zip(name, lp)):
+                    nn = p+'_'+n
+
+                    """
+                    first 3 are lengths, next three are angles
+                    """
+                    if(rid[i] <= 2):
+                        params.add(nn, value=l, lb=l-0.05,
+                                   ub=l+0.05, vary=False)
+                    else:
+                        params.add(nn, value=l, lb=l-1.,
+                                   ub=l+1., vary=False)
```

Comment:
The above section of code looks nearly identical to the section in `if method == 'LeBail':`. It might be better to extract this out of the `if` blocks to avoid code duplication.

Reply:
perhaps put the common functions in a `def _lattice_param_helper()` or something to avoid code duplication.